File: iteration2/code/src/coursetracking/models/Elective.java
package coursetracking.models;

import java.util.ArrayList;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Elective {

    @SerializedName("type")
    @Expose
    public String type;

    @SerializedName("creeditRequirement")
    @Expose
    public int creditRequirement;

    @SerializedName("quota")
    @Expose
    public int quota;

    @SerializedName("courses")
    @Expose
    public ArrayList<Course> courses;
    
    public int getElectiveQuantity(){
        int quantity = courses.size();
        return quantity;
    }

    public int getQuota(){
        return this.quota;
    }

    public boolean isQuotaFull(Course c){
        for(Course course: courses){
            if(course.getCourseCode().equals(c.getCourseCode()) ){
                if(this.quota <= c.getEnrolledSudentsSize()){
                    return true;
                }
            }
        }
        return false;
    }

    public void setSemester(Course c, int semester){
        c.semester = semester;
    }

    public void setNewTerm(){
        for(Course c: courses){
            c.quotaProblemStd = null;
            c.enrolledList = null;
        }
    }
}

File: iteration3/src/model/elective.py
from typing import List

from model.course import Course

class Elective(Course):
    def __init__(self):
        pass

    def __init__(self, course_type: str, credit_requirement: int, quota: int, courses: List[Course]):
        self.__course_type = course_type
        self.__credit_requirement = credit_requirement
        self.__quota = quota
        self.__courses = courses

    def get_elective_quantity(self) -> int:
        return len(self.__courses)

    def is_quota_full(self, cs: Course) -> bool:
        for course in self.__courses:
            if course.__courseCode == cs.__courseCode:
                if self.__quota <= self.getEnrolledSize():
                    return True
        return False

    def set_new_term(self):
        for course in self.__courses:
            course.__stdQuotaProblem = []
            course.__enrolledList = []

    def getQuota(self):
        return self.__quota

    def setSemester(self, course: Course, semester: int):
        course.__semester = semester

    def set_semester(self, course: Course, semester: int):
        course.__semester = semester

    def __str__(self):
        return f"Elective(type={self.__course_type}, credit_requirement={self.__credit_requirement}, quota={self.__quota}, courses={self.__courses})"

    def __repr__(self):
        return f"Elective(type={self.__course_type}, credit_requirement={self.__credit_requirement}, quota={self.__quota}, courses={self.__courses})"